### ISSUE TITLE
Allow definition of a default sort order

### ISSUE THREAD

**Issue by NetTecture** [OData/WebApi#1753]
Right now, elements queried are returned in key order if / when no sort oder is defined in the query.

This is not enough for what we do. In my case, every element is identifie with a GUID as key. Period.

But this is not the normal sort order. The normal sort order is often different, based i.e. on a code number assigned.

The possibility to define, as part of the model, a default sort order, allows returning of rows in a business sensible order even if the actual key for retrieval is based on a random number (like the GUID is).

**Comment by KanishManuja-MS** [comment]
@NetTecture For the framework, it is not possible to make a guess on which properties to sort on for the business case that you have mentioned. With that said, you can always create your own order by query option before you apply it, I will mark this as a feature since we do not allow such configuration for default ordering. 
The Framework uses the keys to order by for stable ordering and that is what you see.

**Comment by NetTecture** [comment]
> For the framework, it is not possible to make a guess

Yeah. Let me see. Title is "allow definition", not "guess". I know. I have no problem defining a DefaultSortFilter as sequence of Sort and ThenSortBy. Heck, now that I think about it this could actually be a global sort order in EF Core actually. I see a LOT of scenarios where this actually can come in handy ;)

And yes, I know how it sorts by default. Only our key, i.e., is a GUID. OTOH we like elements sorted by an internal ID that actually is a consecutive number. it MAY change - but that is an SP reordering all elements ;)

So, yes, this is a feature. One that makes a lot of business sense, because with GUID's the current sort is actually basically - totally random.

**Comment by Galad** [comment]
Hello,
We have a similar use case. Our database tables have an unique Id that is not necesseraly the primary key and is not part of the OData model. We workarounded the default order by disabling the stable order and adding our order to the `ODataQueryOptions` by adding an OrderBy clause where the user defined an order by:

```csharp
            var orderByPropertyNode = GetPropertyNode(typeof(T));
            options.OrderBy.OrderByNodes.Add(orderByPropertyNode);

           OrderByPropertyNode GetPropertyNode(Type type)
            {
                var indexPropertyName = "_Id"; // This is the name of the unique Id property in our model
                var entityType = options.Context.ElementType as IEdmStructuredType;
                // this creates a EDM property that is not actually part of the EDM model
                // the order by clause applies the predicate by using the EDM property name as the CLR type name, and will generate the appropriate expression
                var property = new HiddenProperty(_int32TypeReference, entityType, indexPropertyName);
                return new OrderByPropertyNode(property, OrderByDirection.Ascending);
            }
```

And by just applying the expression on the queryable when there is not user defined order by clause:
```csharp
queryable.OrderBy(entity => entity._Id);
```
This can be achieved by overriding the `EnableQueryAttribute.ApplyQuery` method.

Note that this is little bit of a hack in my opinion and it probably doesn't handle all use cases.

In the end we decided to stick with the order by primary keys by fixing the issue #1762 which blocked us. The reason is that it would prevent us to the the future the skiptoken implementation (#1724).

I'm still interested by having a built-in way to define the order of the records, that could be used a custom skiptoken implementation, with properties that are not part of the model.

**Comment by jeremy-holovacs-sp** [comment]
I also recently ran into a situation where this would be nice to use.
